boxsdk/client/client.py

```diff
@@ -539,6 +539,251 @@ def create_group(self, name):
             response_object=response,
         )
 
+    def retention_policy(self, retention_id):
+        """
+        Initialize a :class:`RetentionPolicy` object, whose box id is retention_id.
+
+        :param retention_id:
+            The box ID of the :class:`RetentionPolicy` object.
+        :type retention_id:
+            `unicode`
+        :return:
+            A :class:`RetentionPolicy` object with the given entry ID.
+        :rtype:
+            :class:`RetentionPolicy`
+        """
+        return self.translator.translate('retention_policy')(session=self._session, object_id=retention_id)
+
+    def file_version_retention(self, retention_id):
+        """
+        Initialize a :class:`FileVersionRetention` object, whose box id is retention_id.
+
+        :param retention_id:
+            The box ID of the :class:`FileVersionRetention` object.
+        :type retention_id:
+            `unicode`
+        :return:
+            A :class:`FileVersionRetention` object with the given retention ID.
+        :rtype:
+            :class:`FileVersionRetention`
+        """
+        return self.translator.translate('file_version_retention')(session=self._session, object_id=retention_id)
+
+    def retention_policy_assignment(self, assignment_id):
+        """
+        Initialize a :class:`RetentionPolicyAssignment` object, whose box id is assignment_id.
+
+        :param assignment_id:
+            The box ID of the :class:`RetentionPolicyAssignment` object.
+        :type assignment_id:
+            `unicode`
+        :return:
+            A :class:`RetentionPolicyAssignment` object with the given assignment ID.
+        :rtype:
+            :class:`RetentionPolicyAssignment`
+        """
+        return self.translator.translate('retention_policy_assignment')(session=self._session, object_id=assignment_id)
+
+    def create_retention_policy(
+            self,
+            policy_name,
+            disposition_action,
+            retention_length,
+            can_owner_extend_retention=None,
+            are_owners_notified=None,
+            custom_notification_recipients=None,
+    ):
+        """
+        Create a retention policy for the given enterprise.
+
+        :param policy_name:
+            The name of the retention policy.
+        :type policy_name:
+            `unicode`
+        :param retention_length:
+            The amount of time in days to apply the retention policy to the selected content.
+            Do not specify for `indefinite` policies, only for `finite` policies.
```

Comment:
This description is no longer accurate; the `retention_length` should be set to `float('inf')` for indefinite policies